src/transcription/utils/utils.ts

```
@@ -44,6 +45,39 @@ export const isRemoteFile = (source: string): boolean => {
   }
 }
 
+/**
+ * Checks if a URL is downloadable.
+ *
+ * @param url - The URL to check.
+ * @returns A Promise that resolves to true if the URL appears to be downloadable; otherwise, false.
+ */
+export const checkIfUrlDownloadable = async (url: string): Promise<boolean> => {
+  try {
+    const response = await axios.get(url, {
+      headers: {
+        Range: 'bytes=0-0',
```

This is not guaranteed to work on all servers. Lots of servers don't support range requests or have them explicitly disabled.

Why do we need to preemptively check if it can be downloaded? Why aren't we just trying to download it and then let it fail if it can't be downloaded?

---

If url is not downloadable, we still receive "succeeded" `status` and error in `ouput` in creating a transcription request

---

Let's replace this check with custom error handling on the response side.

---

Okay
Replaced it